v0/destinations/salesforce/transform.js

```diff
@@ -2,171 +2,184 @@ const get = require("get-value");
 
-  const customKeys = Object.keys(traits);
-  customKeys.forEach(key => {
+    // request configuration will be conditional
+    const leadQueryResponse = await axios.get(leadQueryUrl, {
+      headers: { Authorization: authorizationData.token }
+    });
+
+    let leadObjectId;
     if (
-      !defaultTraits.some(function(k) {
-        return ~k.indexOf(key);
-      }) &&
-      !ignoreMapJson.includes(key)
+      leadQueryResponse &&
+      leadQueryResponse.data &&
+      leadQueryResponse.data.searchRecords
     ) {
-      const val = traits[key];
-      if (val) {
-        payload[`${key}__c`] = val;
+      // if count is greater than zero, it means that lead exists, then only update it
+      // else the original endpoint, which is the one for creation - can be used
+      if (leadQueryResponse.data.searchRecords.length > 0) {
+        leadObjectId = leadQueryResponse.data.searchRecords[0].Id;
       }
     }
-  });
 
-  const response = defaultRequestConfig();
-  const header = {
-    "Content-Type": "application/json",
-    Authorization: authorizationData[0]
-  };
-
-  response.method = defaultPostRequestConfig.requestMethod;
-  response.headers = header;
-  response.body.JSON = { ...customParams, ...payload };
-  response.endpoint = targetEndpoint;
-  response.userId = message.anonymousId;
-  response.statusCode = 200;
+    // add a Lead Object to the response
+    salesforceMaps.push({ salesforceType: "Lead", salesforceId: leadObjectId });
+  }
 
-  return response;
+  return salesforceMaps;
 }
 
 // Function for handling identify events
 async function processIdentify(message, destination) {
+  const responseData = [];
+
   // Get the authorization header if not available
-  // if (!authorizationData) {
   const authorizationData = await getSFDCHeader(destination);
-  // }
-  // start with creation endpoint, update only if Lead does not exist
-  let targetEndpoint = `${authorizationData[1]}/services/data/v${SF_API_VERSION}/sobjects/Lead`;
 
-  // check if the lead exists
-  // need to perform a parameterized search for this using email
-  const { email } = getFieldValueFromMessage(message, "traits");
-
-  const leadQueryUrl = `${authorizationData[1]}/services/data/v${SF_API_VERSION}/parameterizedSearch/?q=${email}&sobject=Lead&Lead.fields=id`;
+  // get salesforce object map
+  const salesforceMaps = await getSalesforceIdFromPayload(
+    message,
+    authorizationData
+  );
 
-  // request configuration will be conditional
-  // POST for create, PATCH for update
-  const leadQueryResponse = await axios.get(leadQueryUrl, {
-    headers: {
-      Authorization: authorizationData[0]
+  // iterate over the object types found
+  salesforceMaps.forEach(salesforceMap => {
+    // finally build the response and push to the list
+    const resp = responseBuilderSimple(
+      message,
+      salesforceMap,
+      authorizationData
+    );
+    if (resp) {
```

Comment:
we should log or throw error in scenarios like this ?
i.e traits not present etc

Reply:
Updated the logic